python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -126,50 +126,87 @@ def _copy_type_metadata(
 
         return type_copied
 
+    def _slice(self: T, arg: slice) -> T:
```

Comment:
If possible, next time let's keep the unchanged function in it's original place and add new function at different lines. Otherwise the diff message will become hard to read.

Reply:
Good idea. I could move it now.